### PR TITLE
RMAC-7234 - Add support for images in web-components

### PR DESCRIPTION
**Pull request by nmmauricio**
In order to have images as resources in the web-components, we need to set the correct URLs.

### CODE REVIEW

File: ViewGenerator/tools/webpack/Rules/Files.ts [hunk 1]
```diff
@@ -24,10 +24,14 @@ const getResourcesRuleSet = (assemblyName?: string, pluginsBase? : string): Rule
                         // Context represents the path of the project being built by webpack, e.g. "C:\Git\Path\to\Project\"
                         //
                         const buildUrl = (url: string, resourceBase: string): string => {
-
+                            let isWebComponentsPackage = false;
```
Comment (alvesmiguel1) [inline ViewGenerator/tools/webpack/Rules/Files.ts:27]: isOutSystemsWebPackage

File: ViewGenerator/tools/webpack/Rules/Files.ts [hunk 2]
```diff
@@ -24,10 +24,14 @@ const getResourcesRuleSet = (assemblyName?: string, pluginsBase? : string): Rule
...
                             let idx: number = url.indexOf(`/${resourceBase}/`);
                             if (idx < 0 && pluginsBase) {                       
                                 idx = url.indexOf(`/${pluginsBase}/`);
+                            } else if (idx < 0 && url.indexOf("/packages/") > 0) {
+                                // web-components packages
```
Comment (alvesmiguel1) [inline ViewGenerator/tools/webpack/Rules/Files.ts:32]: OutSystems web components packages

File: ViewGenerator/tools/webpack/Rules/Files.ts [hunk 3]
```diff
@@ -38,7 +42,11 @@ const getResourcesRuleSet = (assemblyName?: string, pluginsBase? : string): Rule
                                 }
 
                                 // URL (argument) is a relative path and contains the resource base path or the plugin assembly in its content
-                                return url.substring(idx);
+                                if(isWebComponentsPackage) {
+                                    return "../node_modules/webview.plugins/@outsystems/web-" + url.substring(url.indexOf("/packages/") + "/packages/".length);
```
Comment (alvesmiguel1) [inline ViewGenerator/tools/webpack/Rules/Files.ts:46]:
```
let package = url.substring(url.indexOf(PackagesFolder) + PackagesFolder.length);
return `/${pluginsBase}/node_modules/@outsystems/web-${package}`;
```

Comment (alvesmiguel1) [reply ViewGenerator/tools/webpack/Rules/Files.ts:46]: pluginsBase corresponds to ServiceStudio.WebViewImplementation.Framework. It should be always filled when building a project with views

**Review by alvesmiguel1 — CHANGES_REQUESTED** [review]

**Comment by nmmauricio** [comment]
### working icons
image

**Review by alvesmiguel1 — APPROVED** [review]

File: ViewGenerator/tools/webpack/Rules/Files.ts [hunk 4]
```diff
@@ -23,11 +23,50 @@ const getResourcesRuleSet = (assemblyName?: string, pluginsBase? : string): Rule
                         //
                         // Context represents the path of the project being built by webpack, e.g. "C:\Git\Path\to\Project\"
                         //
-                        const buildUrl = (url: string, resourceBase: string): string => {
+                        const PackagesFolder = "/packages/";
+                        const isOutSystemsPackagesWithOldNaming = (url) => {
+                            // means that the package name is (example) "web_feedback-message" or "web_image" instead of "web-feedback-message" or "web-image"
+                            const OutSystemsPackagesWithOldNaming = [
```
Comment (joaompneves) [inline ViewGenerator/tools/webpack/Rules/Files.ts:29]: noooo :(
this is a very specific knowledge that should not be here

Comment (joaompneves) [reply ViewGenerator/tools/webpack/Rules/Files.ts:29]: cant we have another approach?

**Comment by joaompneves** [comment]
@nmmauricio please check my comments and try to address it

File: ViewGenerator/tools/webpack/Rules/Files.ts (hunk 4 above)
Comment (nmmauricio) [reply ViewGenerator/tools/webpack/Rules/Files.ts:29]: I have no clue how to do this differently 😞 

@alvesmiguel1, @mpm-os do you have a different approach?

Comment (mpm-os) [reply ViewGenerator/tools/webpack/Rules/Files.ts:29]: I raised this concern when implementing this. We need to fix the packages with the _ - ...

**Comment by joaompneves** [comment]
is this going to be addressed when?

**Pull request merged**